soccer_vision_3d_rviz_markers/conversion.py

```diff
@@ -0,0 +1,136 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from geometry_msgs.msg import Point
+from soccer_vision_3d_msgs.msg import (
+    Ball, FieldBoundary, Goalpost, MarkingEllipse, MarkingIntersection, MarkingSegment, Obstacle,
+    Robot)
+from soccer_vision_attribute_msgs.msg import Confidence
+from std_msgs.msg import ColorRGBA
+from visualization_msgs.msg import Marker
+
+
+def conf_to_alpha(conf: Confidence) -> float:
+    if conf.confidence == Confidence.CONFIDENCE_UNKNOWN:
+        return 1.0
+    return conf.confidence
+
+
+def ball_to_marker(msg: Ball, diameter: float) -> Marker:
+    marker = Marker()
+    marker.type = Marker.SPHERE
+    marker.pose.position = msg.center
+    marker.scale.x = diameter
+    marker.scale.y = diameter
+    marker.scale.z = diameter
+    marker.color = ColorRGBA(r=1.0, a=conf_to_alpha(msg.confidence))
+    return marker
+
+
+def field_boundary_to_marker(msg: FieldBoundary) -> Marker:
+    marker = Marker()
+    # FieldBoundary has a header field because it's not part of an array. So, copy the header.
+    marker.header = msg.header
+    marker.type = Marker.LINE_STRIP
+    marker.points = msg.points
+    marker.scale.x = 0.02
+    marker.color = ColorRGBA(g=1.0, a=conf_to_alpha(msg.confidence))
+    return marker
+
+
+def goalpost_to_marker(msg: Goalpost) -> Marker:
+    marker = Marker()
+    marker.type = Marker.CYLINDER
+    marker.pose = msg.bb.center
+    marker.scale = msg.bb.size
+    marker.color = ColorRGBA(r=1.0, g=1.0, b=1.0, a=conf_to_alpha(msg.confidence))
+    return marker
+
+
+def marking_ellipse_to_marker(msg: MarkingEllipse) -> Marker:
+    marker = Marker()
+    marker.type = Marker.CYLINDER
+    marker.pose = msg.center
+    marker.scale.x = msg.diameter
+    marker.scale.y = msg.diameter
+    # Set z scale to be very small, but can't be zero because a cylinder needs a height to be
+    # valid. Also, setting this to a lower value makes the cylinder too thin and the marker
+    # starts becoming transparent. 0.005 is a compromise.
+    marker.scale.z = 0.005
+    marker.color = ColorRGBA(r=1.0, g=1.0, b=1.0, a=conf_to_alpha(msg.confidence))
+    return marker
+
+
+def marking_intersection_to_marker(msg: MarkingIntersection) -> Marker:
+    marker = Marker()
+    marker.type = Marker.LINE_LIST
+    marker.pose.position = msg.center
+    for ray in msg.rays:
+        marker.points.append(Point())
+        marker.points.append(Point(x=ray.x * 0.1, y=ray.y * 0.1, z=ray.z * 0.1))
+    marker.scale.x = 0.02  # 0.02m line width
```

Comment:
The configurable parameters correspond to the parameters that are affected by the rules. For example, SPL needs 0.05m line width while MSL would need 0.10m.

With the MarkingEllipse, a width wouldn't make sense because the visualization is a very slim Cylinder (z ~= 0.0).

On the other hand, the width of MarkingIntersection, is just to make the intersection more obvious when overlapping with a MarkingSegment. I could make this a configurable parameter, but I figured that not many teams are going to play around with the value and so left it constant.

If you think the marking intersection line width should be a configurable parameter, I'd suggest we continue the discussion in a new issue.